src/services/schema-form.js

```diff
@@ -314,7 +314,7 @@ angular.module('schemaForm').provider('schemaForm',
             var path = options.path.slice();
             path.push(key);
             
-            var required = schema.required && schema.required.indexOf(key) !== -1;
+            var required = value.required && value.required.length !== 0;
```

Comment:
I tried but it kept bombing out and returning the actual length of the array. It was super bizarre.